src/ui/Scheduler.hpp

```diff
@@ -200,7 +204,10 @@ struct Scheduler {
 public:
     template<typename TScheduler, typename... Args>
     void emplaceScheduler(Args&&... args) {
-        _scheduler = std::make_unique<SchedulerImpl<TScheduler>>(std::forward<Args>(args)..., schedulerThreadPool);
+        using namespace gr::thread_pool;
+        auto cpu = std::make_shared<ThreadPoolWrapper>(std::make_unique<BasicThreadPool>(std::string(kDefaultCpuPoolId), TaskType::CPU_BOUND, 4U, 4U), "CPU");
+        gr::thread_pool::Manager::instance().replacePool(std::string(kDefaultCpuPoolId), std::move(cpu));
```

Comment:
I think this piece of code should be called once in a central place in main() explicitly since it is a global configuration.

Even if we only ever create a single Scheduler throughout the lifetime of the application, other parts of the code might also use the thread pool and then it will become very hard to reason about which part of the code is running on which thread pool.